nixos/modules/services/monitoring/grafana.nix

```diff
@@ -635,20 +637,28 @@ in {
         QT_QPA_PLATFORM = "offscreen";
       } // mapAttrs' (n: v: nameValuePair "GF_${n}" (toString v)) envOptions;
       script = ''
+        set -o errexit -o pipefail -o nounset -o errtrace
+        shopt -s inherit_errexit
+
         ${optionalString (cfg.auth.google.clientSecretFile != null) ''
-          export GF_AUTH_GOOGLE_CLIENT_SECRET="$(cat ${escapeShellArg cfg.auth.google.clientSecretFile})"
+          GF_AUTH_GOOGLE_CLIENT_SECRET="$(<${escapeShellArg cfg.auth.google.clientSecretFile})"
```

Comment:
Just a question. The idea here is to propagate an error linked to a missing file?
What was happening before in case of a missing file and what would happen now?

Reply:
Previously, it would just export an empty variable and continue, now the script fails when trying to read the file.

Reply:
Oh, I didn't read properly the `!= null` above. It makes sense.
Looks good to me.